src/lib/index.ts

```diff
@@ -218,41 +216,30 @@ export class GoogleToken {
       return;
     }
 
-    request(
-        {
-          method: 'post',
-          url: GOOGLE_TOKEN_URL,
-          form: {
-            grant_type: 'urn:ietf:params:oauth:grant-type:jwt-bearer',
-            assertion: signedJWT
-          }
-        },
-        (err2: Error, res: request.RequestResponse, body: any) => {
-          try {
-            body = JSON.parse(body);
-          } catch (e) {
-            body = {};
-          }
-
-          err2 = err2 ||
-              body.error &&
-                  new Error(
-                      body.error +
-                      (body.error_description ? ': ' + body.error_description :
-                                                ''));
-
-          if (err2) {
-            this.token = null;
-            this.tokenExpires = null;
-            callback(err2, null);
-            return;
-          }
-
+    axios
+        .post(GOOGLE_TOKEN_URL, {
+          grant_type: 'urn:ietf:params:oauth:grant-type:jwt-bearer',
+          assertion: signedJWT
+        })
+        .then(r => {
+          const body = r.data;
           this.rawToken = body;
           this.token = body.access_token;
           this.expiresAt = (iat + body.expires_in) * 1000;
           return callback(null, this.token);
+        })
+        .catch(e => {
+          this.token = null;
+          this.tokenExpires = null;
+          const body = e.response.data;
```

Comment:
`e.response` is guaranteed to be non-null non-undefined?

Reply:
Nice catch.  I haven't seen a case where it returns null | undefined, but I switched this up to be a little more defensive just in case.